data_structures/src/data_request.rs

```diff
@@ -204,47 +204,56 @@ impl DataRequestPool {
-                            }
-                            if !present {
-                                log::error!(
+                    match dr_state.stage {
+                        DataRequestStage::REVEAL => {
+                            // When a data request changes from commit stage to reveal stage, it should
+                            // be removed from the "data_requests_by_epoch" map, which stores the data
+                            // requests potentially available for commitment
+                            if let Some(hs) = data_requests_by_epoch.get_mut(&dr_state.epoch) {
+                                let present = hs.remove(dr_pointer);
+                                if hs.is_empty() {
+                                    data_requests_by_epoch.remove(&dr_state.epoch);
+                                }
+                                if !present {
+                                    log::error!(
                                     "Data request {:?} was not present in the \
                                      data_requests_by_epoch map (epoch #{})",
                                     dr_pointer,
                                     dr_state.epoch
                                 );
+                                }
                             }
-                        }
 
-                        if let Some(transaction) = waiting_for_reveal.remove(dr_pointer) {
-                            // We submitted a commit for this data request!
-                            // But has it been included into the block?
-                            let pkh = PublicKeyHash::from_public_key(
-                                &transaction.signatures[0].public_key,
-                            );
-                            if dr_state.info.commits.contains_key(&pkh) {
-                                // We found our commit, return the reveal transaction to be sent
-                                return Some(transaction);
-                            } else {
-                                info!(
-                                    "The sent commit transaction has not been \
-                                     selected to be part of the data request {:?}",
-                                    dr_pointer
+                            if let Some(transaction) = waiting_for_reveal.get(dr_pointer) {
+                                // We submitted a commit for this data request!
+                                // But has it been included into the block?
+                                let pkh = PublicKeyHash::from_public_key(
+                                    &transaction.signatures[0].public_key,
                                 );
-                                debug!(
-                                    "Commit with pkh ({}) removed from the list of commits waiting \
+                                if dr_state.info.commits.contains_key(&pkh) && !dr_state.info.reveals.contains_key(&pkh){
```

Comment:
This `if` should be split into two, otherwise the message in the `else` branch is not correct